### PR TITLE
feat(shorebird_code_push_protocol): Add stripeCustomerId to User

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds the stripeCustomerId field to the User model.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_code_push_protocol/lib/src/models/user.dart [hunk 1]
```diff
@@ -28,4 +29,9 @@ class User {
 
   /// Whether the user is currently a paying customer.
   final bool hasActiveSubscription;
+
+  /// The user's Stripe customer id.
+  ///
+  /// This will be null for users that have never paid.
+  final String? stripeCustomerId;
```
Comment (felangel) [inline packages/shorebird_code_push_protocol/lib/src/models/user.dart:36]: Why is this necessary? I don't think we should expose this publicly imo.

Comment (bryanoltman) [reply packages/shorebird_code_push_protocol/lib/src/models/user.dart:36]: We need this on the server side, but I agree about not exposing it to the client – maybe we should create a different User model for the server?

Comment (felangel) [reply packages/shorebird_code_push_protocol/lib/src/models/user.dart:36]: Yeah I think we should create a UserEntity on the DB side which strictly corresponds to the DB schema and keep that decoupled from the public models.
